tests/integration_tests/functional/test_snapshot_version.py

```diff
@@ -123,3 +151,69 @@ def test_restore_in_past_versions(bin_cloner_path):
 
         exit_code, _, _ = ssh_connection.execute_command("sleep 1 && sync")
         assert exit_code == 0
+
+
+@pytest.mark.skipif(
+    platform.machine() != "x86_64",
+    reason="Not supported yet."
+)
+def test_create_with_past_version(test_microvm_with_ssh, network_config):
+    """Test scenario: restore in previous versions with too many devices."""
+    test_microvm = test_microvm_with_ssh
+
+    # Create and start a microVM with (`FC_V0_23_MAX_DEVICES_ATTACHED` - 1)
+    # network devices.
+    devices_no = FC_V0_23_MAX_DEVICES_ATTACHED - 1
```

Comment:
This is a bit confusing. Why -1 ?

Reply:
The `devices_no` parameter represents the number of network devices we want to attach. When the microVM is spawned, we configure it using `basic_config()`, which attaches a rootfs, so an additional virtio device. This is why we use `FC_V0_23_MAX_DEVICES_ATTACHED - 1` instead of `FC_V0_23_MAX_DEVICES_ATTACHED` for net devices when testing with the maximum number of virtios allowed. 
I will update the comments to better reflect this behavior.